dvc/remote/base.py

```diff
@@ -634,7 +620,7 @@ def exists_with_progress(chunks):
                 results = executor.map(exists_with_progress, chunks)
                 in_remote = itertools.chain.from_iterable(results)
                 ret = list(itertools.compress(checksums, in_remote))
-                progress_callback.finish("")
+                pbar.close()
```

Comment:
it might be closed if `total` is reached but it's safer to explicitly close. Can always use `with` syntax instead.

Also not sure about the control flow logic here - made a small change moving `if self.no_traverse and hasattr(self, "batch_exists")` to the top of the function (56db2ee5)